pyopensprinkler/__init__.py

```diff
@@ -299,6 +306,26 @@ def stations(self):
         """Return stations"""
         return self._stations
 
+    @property
+    def flow_sensor_enabled(self):
+        """Retrieve flow sensor enabled"""
+        return self._get_option("urs") == 2
```

Comment:
This appears deprecated in the API docs and replaced with `snXt` and `snXo`?

Reply:
Updated.
Do you think there should be a firmware version check and warn if below 2.1.9 ? Alternatively, I could add backward compatibility for "urs" etc if that would be acceptable.

Reply:
I'm not entirely sure honestly...we do have this:

```
    @property
    def rain_sensor_enabled(self):
        """Retrieve rain sensor enabled"""
        try:
            return bool(self._get_variable("rs"))
        except KeyError:
            return None
```

Maybe we expand the logic for that to mimic what you're doing and make the check basically be an abstract version.

In addition there seems to be a (relatively) confusing meaning here (note, I'm new to OpenSprinker and don't have any sensors so I'm not entirely sure how they tie together). `enabled` in this context seems to be not only set to type rain but also the thing is `active`. The controller variable `sn1` seems to be indicating if `sn1t>0 AND sn1o=1` but I'm not entirely understanding how these things alter the behavior of OpenSprinkler at this point so whatever light you can shed on that would be great.

Reply:
I think I'm starting to digest this a little better but still not great. I think the current properties need to be renamed to `_active` instead of `_enabled`. Does the flow sensor impact operation at all? Or is just reading data inside OpenSprinkler?

Reply:
I've been a user of OpenSprinkler for around 7-8 years and made quite a few code contributions across Firmware, App and Weather components of OS. Sorry if the following is already known but it might be useful ...

So general terminology is that a sensor is `enabled` when it has been wired up and is listening for the particular condition and is `active` when that condition has been detected. Looking at the current codebase it appears as if the library uses the terms in reverse. Not necessarily a problem just a bit different than I was expecting.

In terms of how the various api options/variables work then the `sn1t` and `sn1o` options let you know what has been configured on that particular gpio pin and the `sn1` option tells you if the condition is active.

Hooking up a Flow Sensor to OpenSprinkler does not effect operation in any way other than providing the user with a sense of water usage via the App. However, with automation such as HA, then it is a very important sensor as it provides a view of water usage that can be helpful if adopting an ETo watering strategy, also for notifying of faults i.e. if it is reading flow whilst idle then it signals that there is a leak in the pipework and if it is reading zero flow when the station is running then the valve is probably fused closed.

It is interesting that the library supports the legacy `rs` option as it means that downstream code needs to understand differences in the OS API versions rather than abstracting that away, i.e. when to call `rain_sensor_` vs `sensor_1_` functions. I also cant see how this library tells downstream code what type of sensor is connected. Maybe that is happening in the HA integration ?